clang/lib/Sema/SemaAttr.cpp

```diff
@@ -85,6 +85,116 @@ void Sema::AddMsStructLayoutForRecord(RecordDecl *RD) {
         MSVtorDispAttr::CreateImplicit(Context, VtorDispStack.CurrentValue));
 }
 
+template <typename Attribute>
+static void addOwnerPointerAttributeIfNotExisting(ASTContext &Context,
+                                                  CXXRecordDecl *Record) {
+  CXXRecordDecl *Canonical = Record->getCanonicalDecl();
+  if (Canonical->hasAttr<OwnerAttr>() || Canonical->hasAttr<PointerAttr>())
+    return;
+
+  Canonical->addAttr(::new (Context) Attribute(SourceRange{}, Context,
+                                               /*DerefType*/ nullptr,
+                                               /*Spelling=*/0));
+}
+
+/// Add gsl::Pointer attribute to std::container::iterator
+/// \param ND The declaration that introduces the name std::container::iterator.
+/// \param UnderlyingRecord The record named by ND.
+void Sema::addDefaultOwnerPointerAttribute(NamedDecl *ND,
```

Comment:
In case this only ever adds Pointer attribute maybe we should get rid of the Owner from the name?